Source/WebCore/loader/SubresourceLoader.cpp

```diff
@@ -291,7 +291,10 @@ void SubresourceLoader::willSendRequestInternal(ResourceRequest&& newRequest, co
                 m_frame->page()->diagnosticLoggingClient().logDiagnosticMessageWithResult(DiagnosticLoggingKeys::cachedResourceRevalidationKey(), emptyString(), DiagnosticLoggingResultFail, ShouldSample::Yes);
         }
 
-        if (!m_documentLoader->cachedResourceLoader().updateRequestAfterRedirection(m_resource->type(), newRequest, options(), originalRequest().url())) {
+        auto originalOrigin = SecurityOrigin::create(redirectResponse.url());
+        m_site = m_documentLoader->cachedResourceLoader().computeFetchMetadataSite(newRequest, m_resource->type(), options().mode, originalOrigin, m_site);
```

Comment:
I am not sure you need to store m_site here.
Maybe you can use the past request Sec-Fetch-Site header value to get it back.

It could also be done in updateRequestAfterRedirection if we were passing the request and not just the request.url()

Reply:
>  Maybe you can use the past request Sec-Fetch-Site header value to get it back.

We have to keep it because the header can go away if an insecure host is used in between a redirect chain.

Reply:
OK

Reply:
Do we have a test that covers this?

Reply:
Yes `fetch/metadata/redirect/` has a few covering this.